include/ck/tensor_operation/gpu/grid/gridwise_layernorm.hpp

```diff
@@ -0,0 +1,368 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2022, Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+
+#include "ck/utility/data_type.hpp"
+#include "ck/utility/reduction_common.hpp"
+#include "ck/utility/reduction_operator.hpp"
+#include "ck/utility/reduction_functions_accumulate.hpp"
+#include "ck/tensor_operation/gpu/block/reduction_functions_blockwise.hpp"
+#include "ck/tensor_operation/gpu/thread/reduction_functions_threadwise.hpp"
+#include "ck/tensor_operation/gpu/thread/threadwise_tensor_slice_transfer.hpp"
+#include "ck/tensor_operation/gpu/element/element_wise_operation.hpp"
+
+namespace ck {
+
+template <typename GridwiseReduction,
+          typename XDataType,
+          typename GammaDataType,
+          typename BetaDataType,
+          typename YDataType,
+          typename AccDataType,
+          typename GridDesc_M_K>
+__global__ void kernel_layernorm(const GridDesc_M_K in_grid_desc_m_k,
+                                 const GridDesc_M_K gamma_grid_desc_m_k,
+                                 const GridDesc_M_K beta_grid_desc_m_k,
+                                 const GridDesc_M_K out_grid_desc_m_k,
+                                 index_t block_group_size,
+                                 index_t num_k_block_tile_iteration,
+                                 AccDataType epsilon,
+                                 const XDataType* const __restrict__ p_x_global,
+                                 const GammaDataType* const __restrict__ p_gamma_global,
+                                 const BetaDataType* const __restrict__ p_beta_global,
+                                 YDataType* const __restrict__ p_y_global)
+{
+    GridwiseReduction::Run(in_grid_desc_m_k,
+                           gamma_grid_desc_m_k,
+                           beta_grid_desc_m_k,
+                           out_grid_desc_m_k,
+                           block_group_size,
+                           num_k_block_tile_iteration,
+                           epsilon,
+                           p_x_global,
+                           p_gamma_global,
+                           p_beta_global,
+                           p_y_global);
+};
+
+template <typename XDataType,
+          typename GammaDataType,
+          typename BetaDataType,
+          typename YDataType,
+          typename AccDataType,
+          typename GridDesc_M_K,
+          index_t BlockSize,
+          index_t MThreadClusterSize,
+          index_t KThreadClusterSize,
+          index_t MThreadSliceSize,
+          index_t KThreadSliceSize,
+          index_t InSrcVectorDim,
+          index_t InSrcVectorSize,
+          index_t GammaSrcVectorDim,
+          index_t GammaSrcVectorSize,
+          index_t BetaSrcVectorDim,
+          index_t BetaSrcVectorSize,
+          index_t OutDstVectorSize,
```

Comment:
Miss ```OutSrcVectorDim```

Reply:
@asroy 
Change in 
cfce1f11ac80f02473f3d031893a597c8e4ae8ff

Similar to softmax, y share the same stride and layout with x in device op.
Therefore, OutSrcVectorDim should be same as XSrcVectorDim.
https://github.com/ROCmSoftwarePlatform/composable_kernel/blob/93c99f3d8701f7c88e7e5389850328f830701017/include/ck/tensor_operation/gpu/grid/gridwise_softmax.hpp#L190-L207

However, I could support YDstVectorDim in gridwise op. But pass same XDstVectorDim in device op

Reply:
ok, that works too.

Please rename ```InSrcVectorDim``` into ```InOutSrcVectorDim``` to clearly indicate that

Reply:
Rename in fe9b4d9a2ea6199374eeddfc615d8e4609b00323